Roachprod azure add machine

smg260 opened the pull request:
This PR makes several azure specific changes to roachtest and roachprod to support more roachtests.

- Add 96 cpu machine type
- Add `s` series machines for `premium/ultra` disks
- Firewall port configuration for more tests and refactor
- Set a default location from `west` to `west2` for AZ
- `apt-get update` for asyncpg

    
Epic: CC-25185
Release note: none

Commit:
roachprod: refactor azure security rules and open ports

This adds some more ports required for various roachtests and
refactors their creation with a helper function.

Epic: CC-25185

Release note: None

Commit:
roachprod: add azure 96cpu machine type

Epic: CC-25185

Release note: None

Commit:
roachtest: apt-get update for asyncpg

This is required so that packages can be found in azure.

Epic: CC-25185

Release note: None

Commit:
roachtest: add support for azure premium/ultra disks

Epic: CC-25185
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111926)
<!-- Reviewable:end -->

Commit:
roachprod: set azure default location westus2

`westus` does not support availability zones, but
`westus2` does. Over 20 roachtests use az.

Epic: CC-25185
Release note: none

DarrylWong commented on pkg/roachprod/vm/azure/flags.go:
```diff
@@ -33,7 +33,7 @@ type ProviderOpts struct {
 var defaultLocations = []string{
 	"eastus",
-	"westus",
+	"westus2",
```
random thought/nit: should we note why we use westus2? Personally I'd be pretty confused but maybe finding the commit message is good enough?

DarrylWong reviewed (approved):
LGTM, someone else should take a look if needed but seems pretty straightforward to me

smg260 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111926)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong and @srosenberg)

___
*[`pkg/roachprod/vm/azure/flags.go` line 36 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/111926#gh-1349278548:-Ng5c654Cx1gin3MkQW1:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/cd825aadd96df409014ae95c279ec324ad75392a/pkg/roachprod/vm/azure/flags.go#L36)):*
<details><summary><i>Previously, DarrylWong wrote…</i></summary><blockquote>

random thought/nit: should we note why we use westus2? Personally I'd be pretty confused but maybe finding the commit message is good enough?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

renatolabs reviewed (commented):
> Reviewed 1 of 1 files at r1, 3 of 3 files at r2, 1 of 1 files at r3, 2 of 2 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111926)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong, @smg260, and @srosenberg)

___
*[`pkg/cmd/roachtest/tests/asyncpg.go` line 85 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/111926#-Ng5lk4J4sM1XK5UJia0:-Ng5lk4J4sM1XK5UJia1:b45zl58) ([raw file](https://github.com/cockroachdb/cockroach/blob/167b2a50f712068af82d95c7192e3bd9bde1225d/pkg/cmd/roachtest/tests/asyncpg.go#L85)):*
> ```Go
> 			ctx, t, c, node, "update apt-get", `sudo apt-get update`,
> 		); err != nil {
> 			t.Fatal(err)
> ```

Curious -- did this fix an Azure specific failure?
___
*[`pkg/roachprod/vm/azure/azure.go` line 897 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/111926#-Ng5jCNJ1S8pbm21seFF:-Ng5jCNJ1S8pbm21seFG:b-z2qm7p) ([raw file](https://github.com/cockroachdb/cockroach/blob/b85e51699a7058fd61e8f814096cd7f9cdaf446d/pkg/roachprod/vm/azure/azure.go#L897)):*
> ```Go
> 
> 	// The names for these are generated in the form Roachtest_<index>_Inbound
> 	genericInbound := []string{"8011", "8081", "9011", "9081-9102", "20011-20016", "27257", "27259-27280", "30258"}
> ```

Where were these port ranges obtained from? Probably a good idea to map them to services in comments if this was obtained manually after test runs, otherwise it will be hard to make sense of this list:

```go
genericInbound := []string{
    "8011", // nginx
    "8081", // foo
}
```

@herkolategan recently added a change to pick a dynamic open port for cockroach. We might have to be more generous with these port ranges with that in mind.

Meta question: do we need to configure the firewall at this level for Azure? AFAIK, we don't have similar configurations for the other clouds. Given that folks might be using roachprod outside of roachtest for their own experimentation, this could be overly restrictive too.


<!-- Sent from Reviewable.io -->

smg260 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111926)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong, @herkolategan, @renatolabs, and @srosenberg)

___
*[`pkg/cmd/roachtest/tests/asyncpg.go` line 85 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/111926#-Ng5lk4J4sM1XK5UJia0:-Ng5ycn7EPYHqmo5JG_c:b-s7gnn9) ([raw file](https://github.com/cockroachdb/cockroach/blob/167b2a50f712068af82d95c7192e3bd9bde1225d/pkg/cmd/roachtest/tests/asyncpg.go#L85)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Curious -- did this fix an Azure specific failure?
</blockquote></details>

It did, as with installing `go` for a cdc. The repository metadata of the vms may be old / out of date. These updates are all over roachtests. At some point it would be nice to give a more structured way of preparing the environment. 
___
*[`pkg/roachprod/vm/azure/azure.go` line 897 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/111926#-Ng5jCNJ1S8pbm21seFF:-Ng5zSZXAOwPnRYptGrs:b3fklv5) ([raw file](https://github.com/cockroachdb/cockroach/blob/b85e51699a7058fd61e8f814096cd7f9cdaf446d/pkg/roachprod/vm/azure/azure.go#L897)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Where were these port ranges obtained from? Probably a good idea to map them to services in comments if this was obtained manually after test runs, otherwise it will be hard to make sense of this list:

```go
genericInbound := []string{
    "8011", // nginx
    "8081", // foo
}
```

@herkolategan recently added a change to pick a dynamic open port for cockroach. We might have to be more generous with these port ranges with that in mind.

Meta question: do we need to configure the firewall at this level for Azure? AFAIK, we don't have similar configurations for the other clouds. Given that folks might be using roachprod outside of roachtest for their own experimentation, this could be overly restrictive too.
</blockquote></details>

When creating the vms, there are default deny all rules in place which are probably configurable, but these rules are only required for processes connecting to the VMs from the internet.  This is really just building on what was already there. 

The ports are from a number of roachtests, many related to multitenant, especially the 20000s which are dynamic tenant SQL ports.  It would be cumbersome to map them all - much of it was parsed from error logs + ide search. 

We probably don't need them and I don't think we restricting connections for GCE or AWS, but it never hurts to err on the side of caution. Either way, these rules are easily modified, and we can revisit removing the completely if it makes sense. 

I've added an explanatory comment


<!-- Sent from Reviewable.io -->

srosenberg commented on pkg/roachprod/vm/azure/azure.go:
```diff
@@ -854,6 +854,77 @@ func (p *Provider) createNIC(
+	// The mapped roachtests are not exhaustive, and at some point will be
+	// cumbersome to keep adding exceptions for. We may need to consider removing
+	// all rules.
+	genericInbound := []string{
```
`workload` has another range for prom metrics: `2112-2120` and `33333` for `pprof` :) (It's in `pkg/workload/cli/run.go`.)

smg260 replied:
Thanks - updated!

smg260 commented:
bors r=darrylwong,srosenberg
